### PR TITLE
[Databricks] Fix abortion state transitions

### PR DESCRIPTION
**Pull request by alonmr**
https://jira.iguazeng.com/browse/ML-5517
Some changes to make sure that databricks runs do not go from aborting to error before moving to aborted

### CODE REVIEW

File: mlrun/runtimes/constants.py [hunk 1]
```diff
@@ -162,6 +162,7 @@ def terminal_states():
     def error_states():
         return [
             RunStates.error,
+            RunStates.aborting,
```
Comment (liranbg) [inline mlrun/runtimes/constants.py:165]: this is not true, aborting is a transitional state, not error state

File: mlrun/runtimes/local.py [hunk 2]
```diff
@@ -487,6 +488,11 @@ def exec_from_params(handler, runobj: RunObject, context: MLClientCtx, cwd=None)
             else:
                 val = handler(**kwargs)
             context.set_state("completed", commit=False)
+        except mlrun.errors.MLRunTaskCancelledError as exc:
+            logger.warning("Run was aborted", err=err_to_str(exc))
+            # Run was aborted, the state run state is updated by the abort job, no need to commit again
+            context.set_state(mlrun.runtimes.constants.RunStates.aborted, commit=False)
+            commit = False
```
Comment (liranbg) [inline mlrun/runtimes/local.py:495]: looking down, logger returns to old level - dont you want to do it as well? perhaps we need to conform both behavior

**Review by liranbg — CHANGES_REQUESTED** [review]

File: mlrun/runtimes/local.py (hunk 2 above)
Comment (alonmr) [reply mlrun/runtimes/local.py:495]: the set logger level is duplicated with one at the end of the function. I have merged them with try, finally clause

File: mlrun/runtimes/constants.py (hunk 1 above)
Comment (alonmr) [reply mlrun/runtimes/constants.py:165]: I think that aborting should be considered an error state. Think that if you want to check if a run is in error state you would also want to know if its aborting so its either that or do
```
run.state in (error_states + [aborting])
```
everywhere

**Review by liranbg — APPROVED** [review]

**Pull request merged**